From 54b5c56cb2e536eec0c0e8e5b1764d60d8896486 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 24 Jul 2017 13:35:40 +0800
Subject: [PATCH 2199/5242] MLK-16075-4 gpu: imx: dpu: prv: Add
 YUV422UPSAMPLINGMODE reg field definitions

commit  271b720852b90740ee4b174237c873f393035b4b from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds YUV422UPSAMPLINGMODE register field definitions.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-prv.h |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/imx/dpu/dpu-prv.h b/drivers/gpu/imx/dpu/dpu-prv.h
index da37aa8..a56e961 100644
--- a/drivers/gpu/imx/dpu/dpu-prv.h
+++ b/drivers/gpu/imx/dpu/dpu-prv.h
@@ -91,6 +91,12 @@ enum linemode {
 #define DELTAY_MASK			0xFC0000
 #define DELTAX(x)			(((x) & 0x3F) << 12)
 #define DELTAY(y)			(((y) & 0x3F) << 18)
+#define YUV422UPSAMPLINGMODE_MASK	BIT(5)
+#define YUV422UPSAMPLINGMODE(m)		(((m) & 0x1) << 5)
+typedef enum {
+	YUV422UPSAMPLINGMODE__REPLICATE,
+	YUV422UPSAMPLINGMODE__INTERPOLATE,
+} yuv422upsamplingmode_t;
 #define SHDTOKGEN			BIT(0)
 #define FETCHTYPE_MASK			0xF
 
-- 
1.7.9.5

